pkg/params/lister.go

```diff
@@ -0,0 +1,280 @@
+// Copyright 2018 The ksonnet authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package params
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"sort"
+
+	"github.com/google/go-jsonnet/ast"
+	"github.com/ksonnet/ksonnet-lib/ksonnet-gen/astext"
+	"github.com/ksonnet/ksonnet-lib/ksonnet-gen/printer"
+	"github.com/ksonnet/ksonnet/pkg/app"
+	"github.com/ksonnet/ksonnet/pkg/util/jsonnet"
+	"github.com/pkg/errors"
+)
+
+// Entry is parameter entry. It uses to hold output when listing
+// parameters.
+type Entry struct {
+	// ComponentName is the component that owns this entry.
+	ComponentName string
+	// ParamName is the name of the parameter.
+	ParamName string
+	// Value is the value of the parameter.
+	Value string
+}
+
+// Lister lists parameters.
+type Lister struct {
+	// Destination is the cluster details.
+	Destination app.EnvironmentDestinationSpec
+	// AppRoot is the root path for the application.
+	AppRoot string
+
+	createEntry func(id string, object *astext.Object) ([]Entry, error)
+}
+
+// NewLister creates an instance of Lister.
+func NewLister(appRoot string, destination app.EnvironmentDestinationSpec) *Lister {
+	ec := newEntryCreator()
+
+	return &Lister{
+		AppRoot:     appRoot,
+		Destination: destination,
+
+		createEntry: ec.Create,
+	}
+}
+
+// List lists parameters in a slice of Entry.
+func (l *Lister) List(r io.Reader, componentName string) ([]Entry, error) {
```

Comment:
A comment on the expected input format (example jsonnet) would be useful for the uninitiated.